src/data/library.ts

```diff
@@ -56,27 +61,27 @@ export class Library {
       return acc;
     }, [] as (readonly NonDeleted<ExcalidrawElement>[])[]);
 
-    Library.saveLibrary([...existingLibraryItems, ...filtered]);
+    this.saveLibrary([...existingLibraryItems, ...filtered]);
   }
 
-  static loadLibrary = (): Promise<LibraryItems> => {
+  loadLibrary = (): Promise<LibraryItems> => {
     return new Promise(async (resolve) => {
-      if (Library.libraryCache) {
-        return resolve(JSON.parse(JSON.stringify(Library.libraryCache)));
+      if (this.libraryCache) {
+        return resolve(JSON.parse(JSON.stringify(this.libraryCache)));
       }
 
       try {
-        const data = localStorage.getItem(STORAGE_KEYS.LOCAL_STORAGE_LIBRARY);
-        if (!data) {
+        const libraryItems = this.app?.libraryItemsFromStorage;
```

Comment:
this needs to be reset upon read otherwise you won't be able to clear the library in the same session

Reply:
I didn't get this, the user will be able to clear the library as the host takes care of it, could you explain more?

Reply:
It affects a case where the `libraryCache` is unset and `loadLibrary()` inits it again from the initial `this.app.libraryItemsFromStorage`.

repro:

1. import library
2. reload browser
3. clear library
4. close library
5. open library
6. => the library items are back

It can be fixed in several ways: e.g. never unsetting the `libraryCache`.

Reply:
oh yeah good catch, yes I think we can stop clearing the cache now since we are reading from initial data so we should read from it only once